enhancements/cluster-logging/fine-grained-per-namespace-logs-access.md

```diff
@@ -0,0 +1,249 @@
+  name: restricted-cluster-admin-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: restricted-cluster-admin
+subjects:
+- kind: Group
+  name: restricted-cluster-admin-group
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Now the cluster admin can allow access to specific namespaces to the `restricted-cluster-admin-group` (or to a specific user in the group) by creating a RoleBinding with the `logging-application-logs-reader`, e.g.
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: special-perm-restricted-cluster-admin-binding
+  namespace: desired-namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: logging-application-logs-reader
+subjects:
+- kind: Group # or User
+  name: restricted-cluster-admin-group # or the User's name
+  apiGroup: rbac.authorization.k8s.io
+```
+
+All of the resources created will not be managed by the operator once they are created. This will allow the cluster admin to have control over how and when they are used.
+
+### API Extensions
+
+We will be modifying the `ClusterLogging` CRD managed by the Cluster Logging Operator by adding a new field to the CRD called `advancedLogsAccess`. This new field is a boolean with a default value of false.
+This new field will be used to enable/disable the fine grained access to logs for the LokiStack logstore.
+
+### Implementation Details/Notes/Constraints 
+
+Here's the modification for the `ClusterLogging` CRD:
+
+```go
+// LokiStackStoreSpec is used to set up cluster-logging to use a LokiStack as logging storage.
+// It points to an existing LokiStack in the same namespace.
+type LokiStackStoreSpec struct {
+	// Name of the LokiStack resource.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// Enable fine grained control over access to logs.
+	//
+	// +nullable
+	// +optional
+  AdvancedLogsAccess bool `json:"advancedLogsAccess"`
+}
+```
+
+Here's an example of a `ClusterLogging` definition with advanced logs access enabled:
+
+```yaml
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogging
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  managementState: Managed
+  logStore:
+    type: lokistack
+    lokistack:
+      name: lokistack-dev
+      advancedLogsAccess: true
+  collection:
+    type: vector
+```
+
+### Risks and Mitigations
+
+- Privilege escalation, ie, a restricted cluster admin being able to grant themselves the ability to access logs they should not be able to access via an RBAC modification, is something cluster admins should be aware of and manage themselves.
+
+Other risks TBD.
+
+### Drawbacks
+
+This proposition assumes that once the advanced logs access is enabled and RBAC resources are created by the operator, the resources are no longer managed by the operator. This means that the remaining responsibilities, such as keeping the list of rules in `restricted-cluster-admin` up to date, will fall onto the cluster admins.
```

Comment:
I believe by modifying our approach we defer these responsibilities entirely to who should control it, a cluster admin.  The operator may not need to modify any rbac